fms/utils/serialization.py

```diff
@@ -109,25 +142,41 @@ def load_state_dict(
     initial_device: where the state dict will be loaded. if meta, return None.
     """
     if model_path is None or initial_device.type == "meta":
-        return {}
-    # TODO: Add support for tp-sharding a non-sharded state dict.
-    if (
-        distributed_strategy == "tp" or checkpoint_sharding == "tp"
-    ) and distributed_strategy != checkpoint_sharding:
-        raise ValueError(
-            f"TP-sharded models are currently only compatible with TP-sharded checkpoints. Attempting to load {checkpoint_sharding} to {distributed_strategy}"
-        )
+        return {}  # type: ignore
```

Comment:
what happened here? this mypy hint wasn't needed on main

Reply:
it thinks {} is Mapping[Any, Any] and it is unhappy about it because it should be Mapping[str, Any]

Reply:
hmm... weird that it worked on main but I think you can also add the type hint for Mapping[str,Any] here. Might have to do something weird like `result:Mapping[str,Any] = {}; return result` though.